### ISSUE TITLE
Whitebox fails to open certain files with non ascii characters in file path.

### ISSUE THREAD

**Issue by purple-blog** [WhiteBox-Systems/whitebox#29]
### What happened?

**Actual behaviour**:
Whitebox is unable to perform certain file based operations on file paths that contain non ascii characters. This behavior is specific to the windows version of whitebox.
**Expected behaviour**:
Whitebox should work with non ascii file paths.

### System/app details

WhiteBox version - 0.122.0
WhiteBox commit  - 395511911a4c91c17e672c059266bf815b254c34

### How to reproduce the error

**Steps to reproduce:**
  1. Place whitebox and its respective program parts into a directory containing a non ascii character such as `é`.
  2. Run whitebox launcher executable or whitebox.exe directly.
  3A. Console should report error `compiler_input error: Couldn't load the library at [Path]`
  3B. A related error occurs when attempting to open the readme.pdf file from within the whitebox program.

**Does the error happen consistently with the above steps?**
Always

### Crash/log attachments & additional info

This issue occurs for windows users who set their username with non ascii characters, such as `vhébert`.

A quick disassembly would seem to suggest that whitebox makes calls to `WideCharToMultiByte` in order to convert UTF16 file paths into UTF8 to perform the following string concatenations with a printf family function.
```c
// readme.pdf case.
sprintf(ManualPath, "%s \"%.*s\/readme.pdf\"", "start /B \"\"", -1, WorkingPath);

// msvcrt_shim.dll case.
sprintf(ShimDLLPath, "\"%.*s\/res\/msvcrt_shim.dll\"", -1, WorkingPath);
```
However there appears to be no instance in which a call to `MultiByteToWideChar` is issued in order to convert either UTF8 paths back into a valid UTF16 path. Windows only understands how to work with UTF16 file paths, not UTF8. For this reason, when whitebox attempts to load the concatenated UTF8 paths on my workstation, it would appear to look something such as the following: `C:\Users\vh├⌐bert\...` when it should be formatted as `C:\Users\vhébert\...`.

I can work around this issue by simply relocating whitebox into a different directory with ascii file paths. With that being said, whitebox should be able to work within directories that contain non ascii file paths.

**Comment by azmr** [comment]
Thanks for the detailed report. I'll have a look into it.

**Comment by azmr** [comment]
This should now be fixed for the next release (commit `5612b777`), along with better handling of wide strings in general.
I suspect there may still be some remaining unhandled edge-cases...
